### PR TITLE
PMM-3705 set correct version for packages during the build

### PR DESCRIPTION
**Pull request by hors**

### CODE REVIEW

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-264-b21c5a6
client docker - perconalab/pmm-client-fb:PR-264-b21c5a6
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm2-client/pmm2-client-PR-264-b21c5a6.tar.gz

**Comment by adivinho** [comment]
Could u prepare a FB with the correct pmm-server-packaging branch?

Screenshot_20181010_175636

**Comment by hors** [comment]
> Could u prepare a FB with the correct pmm-server-packaging branch?
> Screenshot_20190509_112439

yes, sure sorry that I did not provide it: https://github.com/Percona-Lab/pmm-submodules/pull/262

**Comment by hors** [comment]
> @hors Please make sure that binaries that are being built with `make release` receive the full version (`2.0.0-beta1`, or really just anything that is in `VERSION` file) via `PMM_RELEASE_VERSION` environment variable, not awk'ed one.

thank you for comment, I have checked and fixed it.

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-264-b9a1c94
client docker - perconalab/pmm-client-fb:PR-264-b9a1c94
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm2-client/pmm2-client-PR-264-b9a1c94.tar.gz

**Comment by AlekSi** [comment]
https://github.com/Percona-Lab/pmm-submodules/blob/0ca1a447e1ae01ef4cef05aa1ccfe8f27540281b/build/bin/vars#L5-L12

Do we still need this logic given that we build from `PMM-2.0` branch?

File: build/bin/build-client-packages [hunk 1]
```diff
@@ -180,7 +180,7 @@ get_sources(){
     REVISION=$(git rev-parse --short HEAD)
     git reset --hard
     #
-    VERSION=$(cat VERSION)
+    VERSION=$(cat VERSION | awk -F'-' '{print $1}')
```
Comment (AlekSi) [inline build/bin/build-client-packages]: Why not use `vars` file?

File: build/bin/vars [hunk 2]
```diff
@@ -2,7 +2,8 @@ bin_dir=$(cd $(dirname $0); pwd -P)
 root_dir=$(cd $(dirname $0)/../..; pwd -P)
 tmp_dir=${root_dir}/tmp
 branch=$(git rev-parse --abbrev-ref HEAD)
-real_pmm_version=$(cat ${root_dir}/VERSION)
+real_pmm_version=$(cat ${root_dir}/VERSION | awk -F'-' '{print $1}')
+real_pmm_release=$(cat ${root_dir}/VERSION | awk -F'-' '{print $2}')
```
Comment (AlekSi) [inline build/bin/vars]: A comment should be present explaining the difference between version and release, and why we awk.

File: build/bin/build-client-binary [hunk 3]
```diff
@@ -27,13 +27,19 @@ gobuild_component() {
     else
         build_command="go build -o \${binary_dir}/${result_file} ./src/${extract_path}/${component_path}"
     fi
+
+    full_pmm_version=${pmm_version}
```
Comment (AlekSi) [inline build/bin/build-client-binary]: Why? Add comment.

**Comment by AlekSi** [comment]
And guys. `pmm_version`, `full_pmm_version`, `real_pmm_version` – really?

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-264-9c0449f
client docker - perconalab/pmm-client-fb:PR-264-9c0449f
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm2-client/pmm2-client-PR-264-9c0449f.tar.gz

File: build/bin/vars [hunk 4]
```diff
@@ -2,9 +2,15 @@ bin_dir=$(cd $(dirname $0); pwd -P)
...
+
+# In VERSION file we can have numeric value like '2.0.0' as well as
+# alphanumeric value like '2.0.0-alpha3' which we can not be used entirely
+# e.g. in Version directive in spec files. That is why we split it
+# in two values.
+real_pmm_version=$(cat ${root_dir}/VERSION | awk -F'-' '{print $1}')
+real_pmm_release=$(cat ${root_dir}/VERSION | awk -F'-' '{print $2}')
 if [ -z "${pmm_version}" ]; then
-    if [[ ${branch} == "master" ]]; then
+    if [[ ${branch} == 'PMM-2.0' ]]; then
```
Comment (AlekSi) [inline build/bin/vars]: That condition was always false, and we were fine with it. Now it will be true for PMM-2.0 branch. Is it what we desire?

Comment (hors) [reply build/bin/vars]: For master branch we have the same approach. Why do we need to change it for 'PMM-2.0'?

Comment (AlekSi) [reply build/bin/vars]: Because we want PMM 2.0 to better than PMM 1.x. Including build scripts.

Comment (EvgeniyPatlan) [reply build/bin/vars]: @AlekSi want better - requires refactoring. And refactoring should be done separately from other tasks

Comment (AlekSi) [reply build/bin/vars]: See https://github.com/Percona-Lab/pmm-submodules/pull/267

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-264-e41804d
client docker - perconalab/pmm-client-fb:PR-264-e41804d
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm2-client/pmm2-client-PR-264-e41804d.tar.gz

**Pull request merged**